lib/node_modules/@stdlib/fs/resolve-parent-paths/lib/async.js

```diff
@@ -284,17 +283,17 @@ function each( paths, dir, done ) {
 	var count;
 	var child;
 	var spath;
-	var FLG = 0; // flag to track if we are done traversing a directory level
+	var FLG = false; // flag to check if we are done resolving all paths
 	var out;
 	var i;
 
 	count = 0;
-	out = filled( null, paths.length );
+	out = [];
 
 	// Start at a base directory and continue moving up through each parent directory...
 	for ( i = 0; i < paths.length; i++ ) {
 		spath = resolve( dir, paths[ i ] );
-		exists( spath, getCallback( i, spath ) );
+		exists( spath, getCallback( i, spath, dir ) );
```

Comment:
You need to be careful doing async execution like this, as many simultaneous async requests can trigger filesystem errors (e.g., too many file descriptors open, etc).

Reply:
It's possible that `exists` won't trigger this error. Not sure. In `fs/read-file-list`, we have to perform error handling to handle the possibility of file system errors: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/fs/read-file-list/lib/async.js. Hence, my rationale for saying that it may be more straightforward to perform async execution sequentially (not sync!), rather than concurrently.

Reply:
Do you mean how other modes are implemented?

Reply:
Afaik, the `access` syscall doesn't open any file descriptors, just checks permissions?

Reply:
Okay. Then leaving this one as is, with multiple concurrent requests, seems fine.